packages/react-server-cli/src/compileClient.js

```diff
@@ -165,6 +166,13 @@ const packageCodeForBrowser = (entrypoints, outputDir, outputUrl, hot, minify, l
 				},
 			],
 		},
+		resolve: {
+			alias: {
+				// These need to be singletons.
+				"react"        : callerDependency("react"),
```

Comment:
Hmm... yeah, with npm 2 that may wind up being necessary.  In npm 3 land, though, those _should_ be accounted for once these top level deps are handled.

I'd say let's only add others on an as-needed basis.